src/main/kotlin/org/jitsi/nlj/util/PacketInfoQueue.kt

```diff
@@ -17,19 +17,27 @@
 package org.jitsi.nlj.util
 
 import org.jitsi.nlj.PacketInfo
+import org.jitsi.utils.concurrent.SafeExecutor
+import org.jitsi.utils.queue.Executor
 import org.jitsi.utils.queue.PacketQueue
-import java.util.concurrent.ExecutorService
 
 /**
  * A [PacketInfo] queue. We do not want to use the copy functionality, which is why the related
  * methods always throw.
  */
 class PacketInfoQueue(
     id: String,
-    executor: ExecutorService,
+    executor: SafeExecutor,
     handler: (PacketInfo) -> Boolean,
     capacity: Int = 1024
-) : PacketQueue<PacketInfo>(capacity, false, null, id, handler, executor) {
+) : PacketQueue<PacketInfo>(
+    capacity,
+    false,
+    null,
+    id,
+    PacketHandler<PacketInfo> { p -> handler.invoke(p) },
```

Comment:
Would it be possible to add methods to support the old constructor params here?

Reply:
Sure